IntelliTect.TestTools.TestFramework/IntelliTect.TestTools.TestFramework.Tests/TestCaseTests/TestFailureTests.cs

```diff
@@ -0,0 +1,46 @@
+﻿using IntelliTect.TestTools.TestFramework.Tests.TestData.Dependencies;
+using IntelliTect.TestTools.TestFramework.Tests.TestData.TestBlocks;
+using System;
+using Xunit;
+
+namespace IntelliTect.TestTools.TestFramework.Tests.TestCaseTests
+{
+    public class TestFailureTests
+    {
+        [Fact]
+        public void TestFailureThrowsImmediatelyWithOriginalException()
+        {
+            TestBuilder builder = new();
+            TestCase tc = builder
+                .AddDependencyInstance(false)
+                .AddDependencyInstance("Testing")
+                .AddTestBlock<ExampleTestBlockWithBoolReturn>()
+                .AddTestBlock<ExampleTestBlockWithMultipleDependencies>(1)
+                .Build();
+
+            var ex = Assert.Throws<TestCaseException>(() => tc.Execute());
+            Assert.False(tc.Passed);
+            Assert.NotNull(ex.InnerException);
+            Assert.Equal(typeof(DivideByZeroException), ex.InnerException!.GetType());
```

Comment:
```suggestion
            Assert.Equal(typeof(DivideByZeroException), ex.InnerException.GetType());
```
I think this is unnecessary with the notNull check above on ex.InnerException

Reply:
Even better is to leverage xUnit to do these sorts of checks (combines the null and type check into one)
`Assert.IsType<DivideByZeroException>(ex.InnerException)`

Reply:
@BenjaminMichaelis the analyzers don't read Assert.NotNull as fulfilling the null check. :( I added it to keep the red little squigglies at bay.

@Keboo change made in an upcoming push.